staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -734,25 +734,37 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 		typer := newUnstructuredObjectTyper(parameterScheme)
 		creator := unstructuredCreator{}
 
-		validationSchema, err := apiextensionshelpers.GetSchemaForVersion(crd, v.Name)
-		if err != nil {
-			utilruntime.HandleError(err)
-			return nil, fmt.Errorf("the server could not properly serve the CR schema")
+		internalValidationSchema := cached[*apiextensionsinternal.CustomResourceValidation]{
+			Fn: func() (*apiextensionsinternal.CustomResourceValidation, error) {
+				validationSchema, err := apiextensionshelpers.GetSchemaForVersion(crd, v.Name)
+				if err != nil {
+					utilruntime.HandleError(err)
+					return nil, fmt.Errorf("the server could not properly serve the CR schema")
+				}
+				if validationSchema == nil {
+					return nil, nil
+				}
+				internalValidationSchema := &apiextensionsinternal.CustomResourceValidation{}
+				if err := apiextensionsv1.Convert_v1_CustomResourceValidation_To_apiextensions_CustomResourceValidation(validationSchema, internalValidationSchema, nil); err != nil {
+					return nil, fmt.Errorf("failed to convert CRD validation to internal version: %v", err)
+				}
+				return internalValidationSchema, nil
+			},
 		}
-		var internalValidationSchema *apiextensionsinternal.CustomResourceValidation
-		if validationSchema != nil {
-			internalValidationSchema = &apiextensionsinternal.CustomResourceValidation{}
-			if err := apiextensionsv1.Convert_v1_CustomResourceValidation_To_apiextensions_CustomResourceValidation(validationSchema, internalValidationSchema, nil); err != nil {
```

Comment:
oops. This code shouldn't change. Had rebased from another branch that changes this.